Automattic/pocket-casts-ios
# Onboarding: Adds plus purchase modal

| 📘 Project: #500 | 🛫 Depends on: #503 |
|:---:|:---:|

This adds the plus purchase modal with some basic purchase states (will be done in a future PR). This also makes some changes:
- Extracts the common "pill" view to use on the modal windows
- Makes some changes to allow background color customization to the MDC hosting controller
- Adds a "HTMLTextView" to allow inline text and links, this is needed because the easy way to do it is in iOS 15+ :(

## Screenshots

| iPhone SE 14.1 | iPhone 8  - 15.5 | iPhone 14 Pro Max - 16.1 |
|:---:|:---:|:---:|
|<img width="200" src="https://user-images.githubusercontent.com/793774/201452518-359a61c9-0538-4364-98a1-a21b2aeb2fa8.png">|<img width="200" src="https://user-images.githubusercontent.com/793774/201452606-482c7b7b-3b7d-4652-8d49-ca9197272b15.png">|<img width="200" src="https://user-images.githubusercontent.com/793774/201451939-45cef612-d09b-4ab5-b6cc-339955705ab8.png">|

## To test


1. Open the Xcode Project
2. Hit <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>,</kbd> to open the Scheme editor
3. Click the options tab
4. Select the Pocket Casts Configuration.storekit file in the StoreKit configuration dropdown
2. Launch the app with the `onboardingUpdates` Feature Flag enabled
3. Go to the Profile tab
4. Tap the Create Folder button
5. Tap on the Upgrade Now button
6. ✅ Verify the view appears with no free trial information
7. Tap on the monthly option
8. ✅ Verify the option switches
9. Switch back to yearly
10. ✅ Verify the option switches
11. Tap the Privacy Policy link
12. ✅ Verify you are brought to the privacy policy
13. Go back, and tap the Terms and Conditions link
14. ✅ Verify you are brought to the terms
15. Open the `Pocket Casts Configuration.storekit` file
16. Change the introductory offer for both items to Free -> Any Duration you want
17. Restart the app, open the modal again
18. ✅ Verify the view now represents the free trial
19. ✅ Verify the view conforms [with the designs](https://www.figma.com/file/PviUP0aiZctOprDuuYRwpb/Onboarding-%26-Upgrade-v2?node-id=406%3A24844)

## Checklist
